lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1299,9 +1259,13 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
         transValue(II->getOperand(2), BB),
         GetMemoryAccess(cast<MemIntrinsic>(II)), BB);
   case Intrinsic::lifetime_start:
-    return transLifetimeIntrinsicInst(OpLifetimeStart, II, BB);
+    return BM->addLifetimeInst(
+        OpLifetimeStart, transValue(II->getOperand(1), BB),
+        dyn_cast<ConstantInt>(II->getOperand(0))->getSExtValue(), BB);
```

Comment:
Potential problem might be here because LLVM reserves 64 bit for the size while in SPIR-V it only 32 bit.

Reply:
Another problem is that LLVM IR value is *signed*, whereas SPIR-V is *unsigned*.